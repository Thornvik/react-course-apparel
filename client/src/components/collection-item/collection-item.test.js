import React from "react";
import { shallow } from "enzyme";
import { CollectionItem } from "./collection-item.component";

describe("collectionItem component", () => {
  let wrapper;
  let mockAddItem;
  const mockName = "blue hat";
  const mockPrice = "20$";
  const mockImageUrl = "www.testimage.com";

  beforeEach(() => {
    const mockProps = {
      item: {
        name: mockName,
        price: mockPrice,
        imageUrl: mockImageUrl,
      },
      addItem: mockAddItem,
    };

    wrapper = shallow(<CollectionItem {...mockProps} />);
  });

  it("should render collection item component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render imageUrl as a prop on BackgroundImage", () => {
    expect(wrapper.find(".image").prop("style")).toHaveProperty(
      "backgroundImage",
      `url(${mockImageUrl})`
    );
  });

  it("should render name prop in name span", () => {
    expect(wrapper.find(".name").text()).toBe(mockName);
  });

  it("should render price prop in price span", () => {
    expect(wrapper.find(".price").text()).toBe(mockPrice);
  });
});
