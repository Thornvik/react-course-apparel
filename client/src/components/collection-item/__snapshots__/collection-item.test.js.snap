// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collectionItem component should render collection item component 1`] = `ShallowWrapper {}`;
